crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -63,50 +76,117 @@ impl MinimapClickEvent {
     }
 }
 
-impl fmt::Debug for MinimapClickEvent {
+impl fmt::Debug for MinimapPressEvent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?} -> {:?}", self.button, self.position)
     }
 }
 
+#[derive(Event)]
+struct MinimapDragEvent {
+    button: MouseButton,
+    position: Vec2,
+}
+
+impl MinimapDragEvent {
+    fn new(button: MouseButton, position: Vec2) -> Self {
+        Self { button, position }
+    }
+
+    fn button(&self) -> MouseButton {
+        self.button
+    }
+
+    /// Position on the map in 2D flat coordinates (these are not minimap
+    /// coordinates).
+    fn position(&self) -> Vec2 {
+        self.position
+    }
+}
+
+#[derive(Resource)]
+struct DraggingButtons(Vec<MouseButton>);
+
 fn click_handler(
     window_query: Query<&Window, With<PrimaryWindow>>,
     mut input_events: EventReader<MouseButtonInput>,
     hud: HudNodes<With<MinimapNode>>,
     bounds: Res<MapBounds>,
-    mut click_events: EventWriter<MinimapClickEvent>,
+    mut dragging: ResMut<DraggingButtons>,
+    mut click_events: EventWriter<MinimapPressEvent>,
 ) {
     let Some(cursor) = window_query.single().cursor_position() else {
         return;
     };
+
     for event in input_events.iter() {
-        if event.state != ButtonState::Released {
+        if event.state != ButtonState::Pressed {
+            dragging.0.retain(|b| *b != event.button);
             continue;
         }
 
         if let Some(mut relative) = hud.relative_position(cursor) {
+            dragging.0.push(event.button);
             relative.y = 1. - relative.y;
-            let event = MinimapClickEvent::new(event.button, bounds.rel_to_abs(relative));
+            let event = MinimapPressEvent::new(event.button, bounds.rel_to_abs(relative));
             info!("Sending minimap click event {event:?}.");
             click_events.send(event);
         }
     }
 }
 
+fn drag_handler(
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    hud: HudNodes<With<MinimapNode>>,
+    bounds: Res<MapBounds>,
+    dragging: Res<DraggingButtons>,
+    mut drag_events: EventWriter<MinimapDragEvent>,
+) {
+    if dragging.0.is_empty() {
```

Comment:
Instead of using `dragging.0` (at multiple places), we can use Bevy's `[derive(Deref)]` https://docs.rs/bevy/latest/bevy/prelude/derive.Deref.html